### ISSUE TITLE
New resource module : AppFabric

### ISSUE THREAD

**Issue by luigilink** [PowerShell/DscResources#184]
I work on a new resource for SharePoint DSC : Installation of Cumulative update for AppFabric : https://github.com/PowerShell/SharePointDsc/pull/394

This resource is not used only by sharepoint so, it will be interesting to create a new module with this resource and other like Installation of AppFabric binaries, start/stop service, manage configuration, etc..

Thanks in advance for your help.

**Comment by luigilink** [comment]
Here you can find the Url of this new DSC module
https://github.com/luigilink/AppFabricDsc

**Comment by mbreakey3** [comment]
Yes - did you read over the [new resource module submissions page](https://github.com/PowerShell/DscResources/blob/master/NewResourceModuleSubmissions.md#how-to-set-up-the-review-pull-request)? If you're still running into specific issues on this, let me know.

**Comment by mbreakey3** [comment]
Also, make sure it follows the uniform folder format as well as the other [High Quality requirements](https://github.com/PowerShell/DscResources/blob/master/HighQualityModuleGuidelines.md)

**Comment by luigilink** [comment]
To be more closed to SharePointDsc and OfficeOnlineDsc, I copy/paste OfficeOnlineDsc and I adapt it for AppFabricDsc. So I have the same folders tree.

**Comment by luigilink** [comment]
[dev ≡]> git checkout --orphan reviewTarget
Switched to a new branch 'reviewTarget'
[reviewTarget +26 ~0 -0 ~]> git rm -r --cached *
rm '.gitattributes'
rm '.github/CONTRIBUTING.md'
rm '.github/ISSUE_TEMPLATE.md'
rm '.github/PULL_REQUEST_TEMPLATE.md'
rm '.gitignore'
rm 'CHANGELOG.md'
rm 'LICENSE'
rm 'Modules/AppFabricDsc/AppFabricDsc.psd1'
rm 'Modules/AppFabricDsc/DSCResources/MSFT_AFInstall/MSFT_AFInstall.psm1'
rm 'Modules/AppFabricDsc/DSCResources/MSFT_AFInstall/MSFT_AFInstall.schema.mof'
rm 'Modules/AppFabricDsc/DSCResources/MSFT_AFInstall/Readme.md'
rm 'Modules/AppFabricDsc/DSCResources/MSFT_AFInstallCumulativeUpdate/MSFT_AFInstallCumulativeUpdate.psm1'
rm 'Modules/AppFabricDsc/DSCResources/MSFT_AFInstallCumulativeUpdate/MSFT_AFInstallCumulativeUpdate.schema.mof'
rm 'Modules/AppFabricDsc/DSCResources/MSFT_AFInstallCumulativeUpdate/Readme.md'
rm 'Modules/AppFabricDsc/Examples/Resources/AFInstall/1-Install.ps1'
rm 'Modules/AppFabricDsc/Examples/Resources/AFInstallCumulativeUpdate/1-Install.ps1'
rm 'Modules/AppFabricDsc/Modules/AppFabricDsc.Util/AppFabricDsc.Util.psm1'
rm 'README.md'
rm 'Tests/Integration/MSFT_AppFabricServerInstall.Integration.tests.ps1'
rm 'Tests/Integration/MSFT_AppFabricServerInstall.config.ps1'
rm 'Tests/Unit/AppFabricDsc.AFInstall.Tests.ps1'
rm 'Tests/Unit/AppFabricDsc.AFInstallCumulativeUpdate.Tests.ps1'
rm 'Tests/Unit/AppFabricDsc.TestHarness.psm1'
rm 'Tests/Unit/AppFabricDsc.Util.Tests.ps1'
rm 'Tests/Unit/Stubs/AppFabricServer.psm1'
rm 'appveyor.yml'
[reviewTarget +8 ~0 -0 !]> git clean -fxd
Removing .gitattributes
Removing .github/
Removing .gitignore
Removing CHANGELOG.md
Removing LICENSE
Removing Modules/
Removing README.md
Removing Tests/
Removing appveyor.yml
[reviewTarget]> git commit --allow-empty -m 'Start of review'
[reviewTarget (root-commit) 1c76dc3] Start of review
[reviewTarget]> git branch reviewBase
[reviewTarget]> git merge dev
**fatal: refusing to merge unrelated histories**
[reviewTarget]> git merge master
**fatal: refusing to merge unrelated histories**

**Comment by luigilink** [comment]
Impossible to merge dev or master in reviewBase

**Comment by luigilink** [comment]
What's the next step ?

**Comment by mbreakey3** [comment]
It will need to be reviewed by the new module review team, but in the mean time, make sure it's following all of the  [High Quality requirements](https://github.com/PowerShell/DscResources/blob/master/HighQualityModuleGuidelines.md), as that will speed up the review process.

**Comment by johlju** [comment]
Are there still interest of adding the AppFabricDsc resource module to the DSC Resource Kit?
If so I like to help getting this reviewed according to the new process of getting a resource module added to the DSC Resource Kit.
I have opened an issue to update the text on [How to do a new resource module submission](https://github.com/PowerShell/DscResources/issues/356),
but basically if we gonna submit this resource module to DSC Resource Kit we need to make sure it follows the [Quality Governance](https://github.com/PowerShell/DscResources/blob/master/Naming.md#quality-governance).

**Comment by johlju** [comment]
@luigilink are you still interested in linking your resource module to DSC Resource Kit? You will still keep it in your own repo. Let me know if you still want to do this and I will provide the steps to link it here. If you don't want to do this, then I suggest we close this issue. 🙂

**Comment by johlju** [comment]
I have sent in a PR that adds the steps to do a new module submission, until that is accepted, please look here at the proposed texts. 

New module submission: https://github.com/johlju/DscResources/blob/b899c99aa1cab6a481c060c26ab0db5e6e4d4d9c/NewResourceModuleSubmissions.md

Issue template: https://github.com/johlju/DscResources/blob/b899c99aa1cab6a481c060c26ab0db5e6e4d4d9c/.github/ISSUE_TEMPLATE/New_resource_module_submission.md

**Comment by johlju** [comment]
@luigilink have you thought anything more around this? Are you still interested so we should keep the issue open?
